samples/wifi/scan/src/main.c

```diff
@@ -332,8 +333,11 @@ int main(void)
 			return ret;
 		}
 
+		memcpy(params.mac_address.addr, CONFIG_WIFI_MAC_ADDRESS,
```

Comment:
This is wrong. `CONFIG_WIFI_MAC_ADDRESS` is a colon de-limited string, and `params.mac_address.addr` is a byte array.

You need to use a helper (something like reverse of `net_sprint_ll_addr_buf`) to do this translation.